pyartifactory/objects/build.py

```diff
@@ -0,0 +1,206 @@
+        """
+        response = self._get(f"api/{self._uri}")
+        logger.debug("List all builds successful")
+        build_list: BuildListResponse = BuildListResponse.model_validate(response.json())
+        return build_list
+
+    def delete(self, delete_build: BuildDeleteRequest) -> None:
+        """
+        :param delete_build: Model object containing required parameters
+        :return: None
+        """
+        try:
+            for _build_number in delete_build.buildNumbers:
+                response = self._get(
+                    f"api/{self._uri}/{delete_build.buildName}/{_build_number}",
+                )
+                self._response_checker(response)
+        except requests.exceptions.HTTPError as error:
+            http_response: Union[Response, None] = error.response
+            if isinstance(http_response, Response) and http_response.status_code == 404:
+                _http_error = BuildError(**http_response.json())
+                raise BuildNotFoundError("\n".join([_error.message for _error in _http_error.errors]))
+            raise ArtifactoryError from error
+        except ArtifactoryBuildError as error:
+            if error.status == 404:
+                raise BuildNotFoundError(error.message)
+            # at least one build number does not exist
+            raise ArtifactoryError from error
+        else:
+            # all build numbers exist
+            _del = self._post(f"api/{self._uri}/delete", json=delete_build.model_dump())
+            logger.debug(_del.text)
+
+    def build_rename(self, build_name: str, new_build_name: str) -> None:
+        """
+        :param build_name: Build to be renamed
+        :param new_build_name: New Build name
+        :return: None
+        """
+        try:
+            response = self._get(
+                f"api/{self._uri}/{build_name}",
+            )
+            self._response_checker(response)
+        except requests.exceptions.HTTPError as error:
+            http_response: Union[Response, None] = error.response
+            if isinstance(http_response, Response) and http_response.status_code == 404:
+                raise BuildNotFoundError(f"Build {build_name} were not found")
+            raise ArtifactoryError from error
+        except ArtifactoryBuildError as error:
+            raise ArtifactoryError from error
+        else:
+            self._post(f"api/{self._uri}/rename/{build_name}?to={new_build_name}")
+            logger.debug("Build %s successfully renamed to %s", build_name, new_build_name)
+
+    def build_diff(self, build_name: str, build_number: str, older_build_number: str) -> BuildDiffResponse:
+        """
+        :param build_name: Build name to be compared
+        :param build_number: More recent build to be compared
+        :param older_build_number: Starting build to be compared
+        :return: BuildDiffResponse model object containing server response
+        """
+        build_name = build_name.lstrip("/")
```

Comment:
I would do instead `build_name.strip("/")` because we want to remove both leading and trailing /. That comment applies to all lines where `lstrip` is used